### PR TITLE
feat: add cooperative gestures for maps

### PR DESCRIPTION
**Pull request by RafaelSzmarowski**
## Summary

The goal for this PR is to add the option of cooperative gestures for maps, both POI and Choropleth.

This PR needs an update of Maplibre at least up to version 2.2.0 but as the version 2.2.1 fixes some typings issue on Maplibre expressions, it seems better to already update to 2.2.1.

I had troubles to fix the way we defined `FilterExpression`, If you see better solutions, please let me know ! And also this upgrade will have consequences for our users, some types changes will be necessary even if it doesn't appear heavy.

Please check if there is no regression due to the types changes (I've slightly changed the way the functions are made)

There is a change in Chromatic that I can't really explain: https://www.chromatic.com/test?appId=616594e84c1692003af430f3&id=655b8ed6701077f47cbd3365. It doesn't affect many stories so this must be really niche.

(Internal for Opendatasoft only) Associated Shortcut ticket: [sc-44646](https://app.shortcut.com/opendatasoft/story/44646/studio-poi-maps-choropleth-zoom-on-maps-while-holding-control-command).

### Changes

- Update Maplibre to 2.2.1
- Adapt types changes on Maplibre expressions
- Add some CSS to default cooperative gesture screen rules.

## Open discussion

Do you see better typing changes ?

## To be tested

Regression and Chromatic tests 🙏 

## Review checklist

- [ ] Description is complete
- [ ] Commits respect the [Conventional Commits Specification](https://github.com/opendatasoft/ods-dataviz-sdk/blob/main/CONTRIBUTING.md#commit-messages)
- [ ] 2 reviewers (1 if trivial)
- [ ] Tests coverage has improved
- [ ] Code is ready for a release on NPM

### CODE REVIEW

**Review by KevinFabre-ods — APPROVED** [review]
👌

Minor the lint error 😉

File: packages/visualizations/src/components/types.ts [hunk 1]
```diff
@@ -33,3 +35,10 @@ export type PaletteScale = {
 };
 
 export type ColorScale = GradientScale | PaletteScale;
+
+// We expect an array with couples of values and ExpressionInput and for the last element a default ExpressionInput
+export function isGroupByForMatchExpression(
+    value: ExpressionInputType[]
+): value is [ExpressionInputType, ExpressionInputType, ExpressionInputType] {
+    return value.length >= 3 && value.length % 2 === 1;
+}
```
Comment (etienneburdet) [inline packages/visualizations/src/components/types.ts:44]: Will tell that it's 3 values or more with an odd number of values (it's straight maplibre.d.ts without the ExpressionSpecification or array of input cases)
```suggestion
export function isGroupByForMatchExpression(
    value: ExpressionInputType[]
): value is [
    ExpressionInputType,
    ExpressionInputType,
    ExpressionInputType,
    ...ExpressionInputType[],
    ExpressionInputType
] {
    return value.length >= 3 && value.length % 2 === 1;
}
```

**Review by etienneburdet — APPROVED** [review]
LGTM, I left one minor TS comment that led to think the array was exactly 3 elements, appart from that works super well!

File: packages/visualizations/src/components/types.ts (hunk 1 above)
Comment (RafaelSzmarowski) [reply packages/visualizations/src/components/types.ts]: Thanks ! Made the change: https://github.com/opendatasoft/ods-dataviz-sdk/pull/206/commits/69ca6e33d36510cbf958cc7175ac81cddad02b66.

**Pull request merged**